### ISSUE TITLE
ui: add release notes signup to admin UI

### ISSUE THREAD

**Issue by piyush-singh** [cockroachdb/cockroach#43912]
We should add the ability to sign up for CRDB release notes and updates directly to the admin UI. cc @Annebirzin - let's discuss how this will work.

**Comment by Annebirzin** [comment]
Designs are here: https://zpl.io/2pXPNMJ

**Comment by piyush-singh** [comment]
Follow up to the discussion @dhartunian and I had yesterday:

We do not want to add the heavyweight javascript libraries necessary to enable Hubspot forms into our installed product. Further, we can't simply add calls into an API as our that would require setting up vault or something similar to handle secrets like our API keys. This is further complicated by the source availability of crbd.

Instead, we'd like to use a pattern that David had used successfully in CockroachCloud. We will send emails to Hubspot via Segment, our analytics tracking service. You can use segment to fire an `identify` event that will be processed by their automation and inserted into Hubspot. This will give us a way to test this endpoint as well, as we can see the events sent to segment in real time while testing the new email capture.

You can set up the new segment source that I created specifically for the email signups by using this key: `72EEC0nqQKfoLWq0ZcGoTkJFIG9G9SII`.

NOTE: there is an existing analytics client, and we should leave that as is. We would like to set up a new client using the key above ONLY for the email signup field.

**Comment by piyush-singh** [comment]
Andrii's note from sprint planning today: should we extract the key for segment and instead fill that in during our build process?

**Comment by koorosh** [comment]
Reopened. Waiting for two more PRs to be merged: #44856 , #45143

**Comment by dhartunian** [comment]
@koorosh can you also add the following traits to the Segment event?

```
release_notes_sign_up_from_admin_ui: "true"
product_updates: "true"
```
